fuel-merkle/src/common/position.rs

```diff
@@ -137,16 +150,34 @@ impl Position {
         self.parent().sibling()
     }
 
-    /// The left child position.
-    /// See [child](Self::child).
+    /// The child position of the current position given by the direction.
+    /// A child position has a height less 1 than the current position.
+    ///
+    /// A child position is calculated as a function of the current position's
+    /// index and height, and the supplied direction. The left child
+    /// position has the in-order index arriving before the current index;
+    /// the right child position has the in-order index arriving after the
+    /// current index.
+    pub fn child(self, side: Side) -> Self {
+        assert!(self.is_node());
```

Comment:
Should we consider using `debug_assert!` here instead? Realistically, the way we use this class means that we never call `child` on a leaf, but theortically, this can be called in any context, leading to a panic. At the same time, without the `assert!`, we run the risk of overflowing arithmetic. Alternatively, we could make this a fallible function returning `Result`, but that has many downstream effects.

Reply:
This assert already existed in the previous version of the code, and I didn't want to change the extenal behavior here. I feel like this is out-of-scope for this PR. That said, we could at least improve the documentation here.